website/project/model.py

```diff
@@ -1690,18 +1698,14 @@ def rm_pointer(self, pointer, auth):
 
     @property
     def node_ids(self):
-        return [
-            node._id if node.primary else node.node._id
-            for node in self.nodes
-        ]
+        return [x[0] for x in self.to_storage()['nodes'] if x[1] == 'node']
 
     @property
     def nodes_primary(self):
-        return [
-            node
-            for node in self.nodes
-            if node.primary
-        ]
+        ids = [x[0] for x in self.to_storage()['nodes'] if x[1] == 'node']
```

Comment:
Is this faster because it avoids having to load each of the nodes in the list comprehension?